applications/StructuralMechanicsApplication/tests/cpp_tests/test_membrane.cpp

```diff
@@ -0,0 +1,152 @@
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Klaus B. Sautter and Inigo Lopez
+//
+
+// Project includes
+#include "containers/model.h"
+#include "testing/testing.h"
+
+#include "custom_elements/membrane_element.hpp"
+
+namespace Kratos
+{
+namespace Testing
+{
+    void AssignPredifinedDisplacement(Element::Pointer pElement)
+    {
+        const unsigned int number_of_nodes = pElement->GetGeometry().size();
+        const unsigned int dimension = pElement->GetGeometry().WorkingSpaceDimension();
+
+        double displacement = 0.0;
+        for(unsigned int i = 0; i < number_of_nodes; i++){
+            array_1d<double, 3>& disp = pElement->GetGeometry()[i].FastGetSolutionStepValue(DISPLACEMENT);
+            for(unsigned int j = 0; j < dimension; j++){
+                disp[j] = displacement;
+                displacement += 0.1;
+            }
+        }
+    }
+
+    void ComputeRelativeError(Vector& rRelativeError, Element::Pointer pElement, ModelPart& rModelPart)
+    {
+        const unsigned int number_of_nodes = pElement->GetGeometry().size();
+        const unsigned int dimension = pElement->GetGeometry().WorkingSpaceDimension();
+        const unsigned int number_of_dofs = number_of_nodes * dimension;
+        // Set a predifined displacement field to compute the residual
+        AssignPredifinedDisplacement(pElement);
+
+        // Compute RHS and LHS
+        Vector RHS_original = ZeroVector(number_of_dofs);
+        Matrix LHS_original = ZeroMatrix(number_of_dofs,number_of_dofs);
+
+        pElement->Initialize(); // Initialize the element to initialize the constitutive law
+        pElement->CalculateLocalSystem(LHS_original, RHS_original, rModelPart.GetProcessInfo());
+
+        const double delta = 1e-4;
+
+        for(unsigned int i = 0; i < number_of_nodes; i++){
+            for(unsigned int j = 0; j < dimension; j++){
+                // Pinging
+                array_1d<double, 3>& disp = pElement->GetGeometry()[i].FastGetSolutionStepValue(DISPLACEMENT);
+                disp[j] += delta;
+
+                Vector RHS_pinged = ZeroVector(number_of_dofs);
+                Matrix LHS_pinged = ZeroMatrix(number_of_dofs, number_of_dofs);
+                // Compute pinged LHS and RHS
+                pElement->CalculateLocalSystem(LHS_pinged, RHS_pinged, rModelPart.GetProcessInfo());
+
+                for(unsigned int k = 0; k < number_of_dofs; k++){
+                    // Compute the finite difference estimate of the sensitivity
+                    double sensitivity_fd = -(RHS_pinged(k)-RHS_original(k)) / delta;
+                    // Compute the average of the original and pinged analytic sensitivities
+                    double sensitivity_an = 0.5 * (LHS_original(i*dimension + j,k) + LHS_pinged(i*dimension + j,k));
+                    // Compute the relative error between sensitivities
```

Comment:
what do sensitivities have to do with a basic element test?

Reply:
here forward differencing is used to numerically evaluate K and compare it to the one from the element

Reply:
Just adding to Klaus answer: the stiffness matrix is nothing else that the sensitivity of the residual. Validating the implementation of the stiffness matrix by comparing between a finite difference and an analytical approach is a standard check to make sure that the stiffness is correctly implemented.

Reply:
It actually allowed us to catch a bug that @KlausBSautter found and fixed ;)